transfocate/transfocator.py

```diff
@@ -180,3 +185,37 @@ def focus_at(self, value=None, wait=False, timeout=None, **kwargs):
         if wait:
             status_wait(status, timeout=timeout)
         return status
+
+
+class TransfocatorEnergyInterrupt(Exception):
+    """
+    Custom exception returned when input beam energy (user defined
+    or current measured value) changes significantly during
+    calculation
+    """
+    pass
+
+
+def constant_energy(func):
+    """
+    Ensures that requested energy does not change during calculation
+
+    Parameters:
+    transfocator_obj: transfocate.transfocator.Transfocator object
+    energy_type: string
+        input string specifying 'req_energy' or 'beam_energy'
+        to be monitored during calculation
+    """
+    @wraps(func)
+    def with_constant_energy(transfocator_obj, energy_type, *args, **kwargs):
+        try:
+            energy_signal = getattr(transfocator_obj, energy_type)
+        except:
+            raise ValueError("input energy_type not defined")
+        energy_before = energy_signal.get()
+        result = func(transfocator_obj, *args, **kwargs)
+        energy_after = energy_signal.get()
+        if math.isclose(energy_before, energy_after, abs_tol=0.1) == False:
```

Comment:
You don't have to compare here, this is a little more readable

```suggestion
        if not math.isclose(energy_before, energy_after, abs_tol=0.1):
```